geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStore.scala

```diff
@@ -198,6 +198,10 @@ class AccumuloDataStore(val connector: Connector, override val config: AccumuloD
       }
     }
 
+    if (sft.getVisibilityLevel == VisibilityLevel.Attribute && sft.getAttributeCount > 255) {
```

Comment:
*blinks*  Ok, I'd love to know where this came from:)

Reply:
When I was reviewing the changes, I realized that our attribute level vis would not work correctly if you had more than 127 (Byte.MaxValue) attributes, because the attributes are parsed out of the col family without any delimiters [here](https://github.com/locationtech/geomesa/blob/geomesa_2.11-2.3.1/geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/iterators/KryoVisibilityRowEncoder.scala#L62).
I switched the logic there to use unsigned bytes, which will be the same for 0-128 but lets us also go up to 255, then I added the explicit check here so that we'd fail-fast up front instead of getting bad behavior at runtime.
We could fix it by delimiting the col family values, but that would introduce back-compatibility issues, and 255 seems reasonable enough...